app/components/build/UserNameWithEmailPrompt.js

```diff
@@ -0,0 +1,399 @@
+
+    const { message, buttons } = this.getMessages(loading);
+    let buttonContent;
+
+    if (buttons) {
+      buttonContent = (
+        <Media className="mx4 mt2 my3" style={{ marginTop: -5 }}>
+          <Media.Image className="mr2 center" style={{ width: 32 }}>
+            {loading && <Spinner />}
+          </Media.Image>
+          <Media.Description
+            className="flex flex-auto flex-wrap"
+            style={{ margin: -5 }}
+          >
+            {buttons}
+          </Media.Description>
+        </Media>
+      );
+    }
+
+    if (message) {
+      return (
+        <div>
+          <Dropdown
+            width={460}
+          >
+            <span>
+              <svg
+                className="cursor-pointer"
+                style={{
+                  color: '#7eaf25',
+                  width: '1.25em',
+                  height: '1.25em',
+                  margin: '-.175em',
+                  verticalAlign: '-.15em'
+                }}
+              >
+                <circle
+                  cx="50%"
+                  cy="50%"
+                  r="25%"
+                  stroke="currentColor"
+                  strokeWidth="7%"
+                  fill="none"
+                />
+                <circle
+                  cx="50%"
+                  cy="50%"
+                  r="25%"
+                  stroke="currentColor"
+                  strokeWidth="7%"
+                  fill="none"
+                >
+                  <animate
+                    attributeName="r"
+                    begin="2s"
+                    dur="1.5s"
+                    repeatCount="indefinite"
+                    values="25%;25%;45%"
+                  />
+                  <animate
+                    attributeName="opacity"
+                    begin="2s"
+                    dur="1.5s"
+                    repeatCount="indefinite"
+                    values="0;1;.6;.6;0"
+                  />
+                </circle>
+                <circle
+                  cx="50%"
+                  cy="50%"
+                  r="15%"
+                  fill="currentColor"
+                />
+              </svg>
+            </span>
+            <Media align="top" className="mx4 my3">
+              <Media.Image className="mr2 center">
+                <svg width="32px" height="32px" viewBox="0 0 32 32" version="1.1">
+                  <g transform="translate(1, 1)" fill="none" strokeWidth="2" stroke="#7EAF25">
+                    <ellipse cx="15" cy="15" rx="15" ry="15" />
+                    <rect x="7" y="9" width="16" height="12" rx="2" />
+                    <polyline points="7 9 15 15 23 9.00146484" />
+                  </g>
+                </svg>
+              </Media.Image>
+              {message}
+            </Media>
+            {buttonContent}
+            <Media className="mx4 my3">
+              <Media.Image className="mr2 center" style={{ width: 32 }} />
```

Comment:
I'm finding all this `Media` stuff icky? Is there any reason we’re still using it, instead of just using flexbox? This class is getting a bit gnarly with the conditionals, and having variables containing chunks of JSX.